src/quicklist.c

```diff
@@ -752,8 +754,10 @@ void quicklistReplaceEntry(quicklistIter *iter, quicklistEntry *entry,
         if (entry->node->count == 1) {
             __quicklistDelNode(quicklist, entry->node);
         } else {
+            quicklistDecompressNodeForUse(entry->node);
```

Comment:
please add a comment stating why this is needed quicklistInsertAfter screws up what the iterator prepared for quicklistReplaceEntry.

p.s. maybe we should pass a flag to quicklistInsertAfter not to do the compression? in both cases (count==1 and not) we it's wasteful.

Reply:
I've added the `dont_compress` flag to the quicklistNode, please see the top comment.